server/middlewares/authorizeUser.js

```diff
@@ -0,0 +1,20 @@
+import models from '../database/models';
+import responseMessage from '../helpers';
+
+const { UserRole, Role } = models;
+
+export default async (request, response, next, roleArray) => {
+  try {
+    const { id } = request.userData;
+    const userrole = await UserRole.findOne({
+      where: { userId: id },
+      include: [Role]
+    });
+    if (!userrole) responseMessage(response, 403, { message: 'UnAuthorized User. Role not found' });
+    const { roleName } = userrole.Role;
+    if (roleArray.indexOf(roleName) === -1) responseMessage(response, 403, { message: 'UnAuthorized User' });
```

Comment:
What I understand is happening here is that roleName is checked if it exits in the roleArray. What happens when a user with a "user role" tried to do something meant for the "adminRole"

Reply:
The array specifies who can access the route